File: src-tauri/src/commands/projects.rs
// 项目管理相关命令
use serde::{Deserialize, Serialize};
use crate::db;

/// 项目数据结构
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Project {
    pub id: String,
    pub name: String,
    pub description: Option<String>,
    #[serde(rename = "createdAt")]
    pub created_at: String,
    #[serde(rename = "updatedAt")]
    pub updated_at: String,
}

/// 获取所有项目
#[tauri::command]
pub async fn get_projects(app_handle: tauri::AppHandle) -> Result<Vec<Project>, String> {
    db::projects::get_all_projects(&app_handle)
        .map_err(|e| e.to_string())
}

/// 创建项目
#[tauri::command]
pub async fn create_project(
    project: Project,
    app_handle: tauri::AppHandle,
) -> Result<Project, String> {
    db::projects::create_project(&app_handle, &project)
        .map_err(|e| e.to_string())
}

/// 更新项目
#[tauri::command]
pub async fn update_project(
    project: Project,
    app_handle: tauri::AppHandle,
) -> Result<Project, String> {
    db::projects::update_project(&app_handle, &project)
        .map_err(|e| e.to_string())
}

/// 删除项目
#[tauri::command]
pub async fn delete_project(
    project_id: String,
    app_handle: tauri::AppHandle,
) -> Result<(), String> {
    db::projects::delete_project(&app_handle, &project_id)
        .map_err(|e| e.to_string())
}

/// 获取当前选中的项目ID
#[tauri::command]
pub async fn get_current_project(app_handle: tauri::AppHandle) -> Result<Option<String>, String> {
    db::projects::get_current_project(&app_handle)
        .map_err(|e| e.to_string())
}

/// 设置当前项目
#[tauri::command]
pub async fn set_current_project(
    project_id: String,
    app_handle: tauri::AppHandle,
) -> Result<(), String> {
    db::projects::set_current_project(&app_handle, &project_id)
        .map_err(|e| e.to_string())
}

File: src/lib/api/projects.ts
// Tauri API 封装层 - 项目相关
import { invoke } from '@tauri-apps/api/core';
import type { Project } from '@/types/kanban';

// 将后端格式转换为前端类型
function fromBackendProject(data: Record<string, unknown>): Project {
  return {
    id: data.id as string,
    name: data.name as string,
    description: data.description as string | undefined,
    createdAt: new Date(data.createdAt as string),
    updatedAt: new Date(data.updatedAt as string),
  };
}

export const projectsApi = {
  /** 获取所有项目 */
  async getAll(): Promise<Project[]> {
    const result = await invoke<Record<string, unknown>[]>('get_projects');
    return result.map(fromBackendProject);
  },

  /** 创建项目 */
  async create(project: Partial<Project> & { name: string }): Promise<Project> {
    const now = new Date();
    const backendProject = {
      id: project.id || '',
      name: project.name,
      description: project.description || null,
      createdAt: now.toISOString(),
      updatedAt: now.toISOString(),
    };
    const result = await invoke<Record<string, unknown>>('create_project', {
      project: backendProject,
    });
    return fromBackendProject(result);
  },

  /** 更新项目 */
  async update(project: Project): Promise<Project> {
    const backendProject = {
      id: project.id,
      name: project.name,
      description: project.description || null,
      createdAt: project.createdAt instanceof Date ? project.createdAt.toISOString() : project.createdAt,
      updatedAt: new Date().toISOString(),
    };
    const result = await invoke<Record<string, unknown>>('update_project', {
      project: backendProject,
    });
    return fromBackendProject(result);
  },

  /** 删除项目 */
  async delete(projectId: string): Promise<void> {
    await invoke('delete_project', { projectId });
  },

  /** 获取当前项目ID */
  async getCurrentId(): Promise<string | null> {
    const result = await invoke<string | null>('get_current_project');
    return result;
  },

  /** 设置当前项目 */
  async setCurrent(projectId: string): Promise<void> {
    await invoke('set_current_project', { projectId });
  },
};
